src/EntityComponentManager.cc

```diff
@@ -1021,13 +1132,13 @@ void EntityComponentManagerPrivate::CalculateStateThreadLoad()
 {
   // If the entity component vector is dirty, we need to recalculate the
   // threads and each threads work load
-  if (!this->entityComponentsDirty)
+  if (!this->componentTypeIndexDirty)
     return;
 
-  this->entityComponentsDirty = false;
-  this->entityComponentIterators.clear();
-  auto startIt = this->entityComponents.begin();
-  int numComponents = this->entityComponents.size();
+  this->componentTypeIndexDirty = false;
+  this->componentTypeIndexIterators.clear();
+  auto startIt = this->componentTypeIndex.begin();
+  int numComponents = this->componentTypeIndex.size();
```

Comment:
`this->componentTypeIndex.size()` is actually the number of entities, not components (the key in the `componentTypeIndex` map is the entity). This makes me wonder if the documentation/variable naming for the rest of this method needs to be updated :thinking:

Reply:
hah good call, changed the wording in 01be05f